src/Microsoft.Kusto.ServiceLayer/DataSource/KustoDataSource.cs

```diff
@@ -429,7 +429,12 @@ public override async Task<IEnumerable<T>> ExecuteControlCommandAsync<T>(string
             foreach (var autoCompleteItem in completion.Items)
             {
                 var label = autoCompleteItem.DisplayText;
-                completions.Add(AutoCompleteHelper.CreateCompletionItem(label, label + " keyword", label, KustoIntellisenseHelper.CreateCompletionItemKind(autoCompleteItem.Kind), scriptDocumentInfo.StartLine, scriptDocumentInfo.StartColumn, textPosition.Character));
+                var insertText = KustoQueryUtils.EscapeName(autoCompleteItem.DisplayText);
+                var completionKind = KustoIntellisenseHelper.CreateCompletionItemKind(autoCompleteItem.Kind);
```

Comment:
Is completionKind same as autoCompleteItem.Kind?

Reply:
Completion Kind is the kind from the database, it gets translated to the AutoCompleteItem Kind when sent to ADS. I didn't use that because it translates Table -> File which seems odd for the AutoComplete suggestions.

Reply:
I just moved this out of the function and into it's own variable to make it easier to read.